### PR TITLE
feat: Add metrics for pull query request/response size in bytes

### PR DESCRIPTION
**Pull request by vpapavas**
### Description 
Add metrics for pull query request/response size in bytes. I added it only to the `StreamedQueryResource` and not the `WSQueryEndpoint` since in the latter I don't have access to the vertx objects to measure their size. Also, it's not important to measure the size for queries issued from the UI editor.

### Testing done 
Local testing with issuing pull queries and checking the value of the metrics

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/OldApiUtils.java [hunk 1]
```diff
@@ -112,6 +118,9 @@ static void handleOldApiResponse(final Server server, final RoutingContext routi
         response.end(responseBody);
       }
     }
+    pullQueryMetrics
```
Comment (vpapavas) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/OldApiUtils.java:121]: This is the only place where we have access to the response sent over the wire. That's why I had to pass the `pullQueryMetrics` object in here

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java [hunk 2]
```diff
@@ -139,24 +138,19 @@
...
   private final RateLimiter rateLimiter;
-  private final Optional<PullQueryExecutorMetrics> pullQueryMetrics;
 
   public PullQueryExecutor(
       final KsqlExecutionContext executionContext,
       final RoutingFilterFactory routingFilterFactory,
       final KsqlConfig ksqlConfig,
-      final String serviceId,
-      final Time time
+      final String serviceId
```
Comment (vpapavas) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java:146]: Instead of passing the things we want to measure, like the startTime or the requestSize, I figured it's cleaner to pass the pullQueryMetrics. Also, the response cannot be measured from in here

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/PullQueryPublisher.java [hunk 3]
```diff
@@ -34,33 +35,48 @@
...
         subscriber,
-        () -> pullQueryExecutor.execute(query, serviceContext, Optional.of(false), startTimeNanos)
+        () -> {
+          final PullQueryResult result = pullQueryExecutor.execute(
+              query, serviceContext, Optional.of(false), pullQueryMetrics);
+          if (pullQueryMetrics.isPresent()) {
+            //Record latency at microsecond scale
+            final long nowNanos = Time.SYSTEM.nanoseconds();
+            final double latency = TimeUnit.NANOSECONDS.toMicros(nowNanos - startTimeNanos);
+            pullQueryMetrics.get().recordLatency(latency);
```
Comment (vpapavas) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/PullQueryPublisher.java:76]: I am not measuring the request/response here since I don't have access to the HttpServerRequest/HttpServerResponse objects here

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ServerVerticle.java [hunk 4]
```diff
@@ -227,32 +231,34 @@ private void handleQueryRequest(final RoutingContext routingContext) {
 
     final CompletableFuture<Void> connectionClosedFuture = new CompletableFuture<>();
     routingContext.request().connection().closeHandler(v -> connectionClosedFuture.complete(null));
-
-    handleOldApiRequest(server, routingContext, KsqlRequest.class,
+    routingContext.request().bytesRead();
```
Comment (AlanConfluent) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/ServerVerticle.java:234]: Noop line?

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerEndpoints.java [hunk 5]
```diff
@@ -151,17 +156,20 @@ public KsqlServerEndpoints(
   }
 
   @Override
-  public CompletableFuture<EndpointResponse> executeQueryRequest(final KsqlRequest request,
+  public CompletableFuture<EndpointResponse> executeQueryRequest(
```
Comment (AlanConfluent) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerEndpoints.java:159]: Nit: Unnecessary whitespace changes make it harder to use blame.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutorMetrics.java [hunk 6]
```diff
@@ -83,18 +84,24 @@ public void recordRemoteRequests(final double value) {
     this.remoteRequestsSensor.record(value);
   }
 
-  public void recordLatency(final long startTimeNanos) {
+  public void recordLatency(final double value) {
     // Record latency at microsecond scale
-    final long nowNanos = time.nanoseconds();
```
Comment (AlanConfluent) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutorMetrics.java:88]: Why did you remove how this was being done by encapsulating the nowNanos?  It prevented you from having to do that all over.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/OldApiUtils.java [hunk 7]
```diff
@@ -112,6 +120,9 @@ static void handleOldApiResponse(final Server server, final RoutingContext routi
...
+        .ifPresent(pullQueryExecutorMetrics -> pullQueryExecutorMetrics.recordResponseSize(
```
Comment (AlanConfluent) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/OldApiUtils.java:124]: Why not do recordLatency here too?  You can even record the start time here as well.  That way, they don't have to be sprinkled everywhere.  You would still have to handle the web socket since I don't believe it uses this path.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/impl/QueryEndpoint.java [hunk 8]
```diff
@@ -43,34 +44,38 @@
...
     this.pullQueryExecutor = pullQueryExecutor;
+    this.pullQueryMetrics = pullQueryMetrics;
   }
 
   public QueryPublisher createQueryPublisher(
       final String sql, final JsonObject properties,
       final Context context,
       final WorkerExecutor workerExecutor,
       final ServiceContext serviceContext) {
-    final long startTimeNanos = Time.SYSTEM.nanoseconds();
```
Comment (AlanConfluent) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/api/impl/QueryEndpoint.java:66]: This no longer measures the latency for this endpoint.  Why did you remove it?

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/OldApiUtils.java (hunk 7 above)
Comment (AlanConfluent) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/OldApiUtils.java:124]: Ideally we should measure every one of these endpoints in a generic fashion rather than having this special thing for pull queries, but maybe that is for a followup.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java (hunk 2 above)
Comment (AlanConfluent) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutor.java:146]: I think the idea was that PullQueryExecutorMetrics was encapsulated here, but I agree that you can't do everything that way.  Similarly, we cut corners in measuring time by putting that in here to maintain encapsulation.  Andy felt strongly about this, so I made it that way, but I agree that if we want to measure size, it should be moved out.  I think we should measure time in exactly the same manner since we're already pulling PullQueryExecutorMetrics out.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutorMetrics.java (hunk 6 above)
Comment (vpapavas) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutorMetrics.java:88]: I don't understand what you mean here :(

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerEndpoints.java (hunk 5 above)
Comment (vpapavas) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerEndpoints.java:159]: The formatting is off here, when we break arguments in multiple lines, the first one must be in a new line as well

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/impl/QueryEndpoint.java (hunk 8 above)
Comment (vpapavas) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/api/impl/QueryEndpoint.java:66]: Yeah, you are right. I got a bit trigger happy with removing things. Will add it back

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutorMetrics.java (hunk 6 above)
Comment (vpapavas) [reply ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutorMetrics.java:88]: Added it back, you are right

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutorMetrics.java [hunk 9]
```diff
@@ -85,6 +89,8 @@ public void recordRemoteRequests(final double value) {
 
   public void recordLatency(final long startTimeNanos) {
     // Record latency at microsecond scale
+    //this.latencySensor.record(value);
```
Comment (AlanConfluent) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutorMetrics.java:92]: Remove commented lines.

File: ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/execution/PullQueryExecutorMetricsTest.java [hunk 10]
```diff
@@ -56,7 +56,7 @@ public void setUp() {
     when(ksqlEngine.getServiceId()).thenReturn(KSQL_SERVICE_ID);
     when(time.nanoseconds()).thenReturn(6000L);
```
Comment (AlanConfluent) [inline ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/execution/PullQueryExecutorMetricsTest.java:57]: How is this time used if it's not injected in?  I suspect the test won't pass.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutorMetrics.java [hunk 11]
```diff
@@ -46,19 +48,19 @@
...
 
   public PullQueryExecutorMetrics(
       final String ksqlServiceId,
-      final Map<String, String> customMetricsTags,
-      final Time time
+      final Map<String, String> customMetricsTags
   ) {
 
     this.customMetricsTags = Objects.requireNonNull(customMetricsTags, "customMetricsTags");
-    this.time = Objects.requireNonNull(time, "time");
```
Comment (AlanConfluent) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutorMetrics.java:61]: If you leave time injected in, it'll make it easier to test this class.  In fact, I would imagine it would be hard to test if it's using a wall clock.

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/OldApiUtils.java [hunk 12]
```diff
@@ -64,22 +69,30 @@ private OldApiUtils() {
     } else {
       requestObject = null;
     }
-
+    pullQueryMetrics
+        .ifPresent(pullQueryExecutorMetrics -> pullQueryExecutorMetrics.recordRequestSize(
+            routingContext.request().bytesRead()));
+    //Record latency at microsecond scale
+    pullQueryMetrics.ifPresent(pullQueryExecutorMetrics -> pullQueryExecutorMetrics
```
Comment (AlanConfluent) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/api/server/OldApiUtils.java:76]: You want to record this when the response is complete in `handleOldApiResponse`.

**Review by AlanConfluent — COMMENTED** [review]
Just a few last comments.

**Review by AlanConfluent — APPROVED** [review]
